src/test/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/integration/LicenceExpiryIntegrationTest.kt

```diff
@@ -0,0 +1,71 @@
+package uk.gov.justice.digital.hmpps.createandvaryalicenceapi.integration
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.groups.Tuple
+import org.junit.jupiter.api.Test
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.http.MediaType
+import org.springframework.test.context.jdbc.Sql
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.LicenceSummary
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.model.request.MatchLicencesRequest
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.repository.LicenceRepository
+import uk.gov.justice.digital.hmpps.createandvaryalicenceapi.util.LicenceStatus
+
+class LicenceExpiryIntegrationTest : IntegrationTestBase() {
+
+  @Autowired
+  lateinit var licenceRepository: LicenceRepository
+
+  @Test
+  @Sql(
+    "classpath:test_data/seed-licences-for-expiry.sql",
+  )
+  fun `Run licence expiry job`() {
+    webTestClient.post()
+      .uri("/run-expire-licences-job")
+      .accept(MediaType.APPLICATION_JSON)
+      .headers(setAuthorisation(roles = listOf("ROLE_CVL_ADMIN")))
+      .exchange()
+      .expectStatus().isOk
+
+    val inactiveLicences = webTestClient.post()
+      .uri("/licence/match")
+      .bodyValue(MatchLicencesRequest(status = listOf(LicenceStatus.INACTIVE)))
+      .accept(MediaType.APPLICATION_JSON)
+      .headers(setAuthorisation(roles = listOf("ROLE_CVL_ADMIN")))
+      .exchange()
+      .expectBodyList(LicenceSummary::class.java)
+      .returnResult().responseBody
+
+    assertThat(inactiveLicences?.size).isEqualTo(3)
+    assertThat(inactiveLicences)
+      .extracting<Tuple> {
+        Tuple.tuple(it.licenceId, it.licenceStatus)
+      }
+      .contains(
+        Tuple.tuple(2L, LicenceStatus.INACTIVE),
+        Tuple.tuple(5L, LicenceStatus.INACTIVE),
+        Tuple.tuple(6L, LicenceStatus.INACTIVE),
+      )
+
+    val remainingLicences = webTestClient.post()
+      .uri("/licence/match")
+      .bodyValue(MatchLicencesRequest(status = LicenceStatus.IN_FLIGHT_LICENCES))
```

Comment:
This doesn't look right

Reply:
May I ask why? The purpose of this set of assertions is to make sure that the licences we'd expect to be unaffected are still in the statuses they were in originally. The easiest way to do so was to fetch all of the in-flight licences after the method-in-question had been run